src/porepy/params/data.py

```diff
@@ -1,767 +1,253 @@
-
-        """
-        physics = self._get_physics(obj)
-
-        if physics == "flow":
-            self._bc_flow = val
-        elif physics == "transport":
-            self._bc_transport = val
-        elif physics == "mechanics":
-            self._bc_mechanics = val
-        else:
-            raise ValueError(
-                'Unknown physics "%s".\n Possible physics are: %s'
-                % (physics, self.known_physics)
-            )
-
-    def get_bc_flow(self):
-        """ BoundaryCondition object
-        Cell wise permeability, represented as a second order tensor.
-        Solvers should rather access get_tensor().
-        """
-        if hasattr(self, "_bc_flow"):
-            return self._bc_flow
-        else:
-            return BoundaryCondition(self.g)
-
-    bc_flow = property(get_bc_flow)
-
-    def get_bc_transport(self):
-        """ bc.BoundaryCondition
-        Cell wise conductivity, represented as a second order tensor.
-        Solvers should rather access tensor().
-        """
-        if hasattr(self, "_bc_transport"):
-            return self._bc_transport
-        else:
-            return BoundaryCondition(self.g)
-
-    bc_transport = property(get_bc_transport)
-
-    def get_bc_mechanics(self):
-        """ Stiffness matrix, defined as fourth order tensor
-        """
-        if hasattr(self, "_bc_mechanics"):
-            return self._bc_mechanics
-        else:
-            return BoundaryConditionVectorial(self.g)
-
-    bc_mechanics = property(get_bc_mechanics)
-
-    # Boundary value
-
-    def get_bc_val(self, obj):
-        """ Pick out physics-specific boundary condition
-
-        Discretization methods should access this method.
-
-        Parameters:
-
-        obj : Solver
-            Discretization object. Should have attribute 'physics'.
-
-        Returns:
+"""
+Utility methods for handling of dictionaries.
+TODO: Improve/add/remove methods based on experience with setting up problems using the
+new Parameters class.
+"""
 
-        BoundaryCondition, for
-            flow/pressure equation y if physics equals 'flow'
-            transport equation if physics equals 'transport'
-            elasticity if physics equals 'mechanics'
-            If the BoundaryCondition is not specified, Neumann conditions will
-            be assigned.
-        """
-        physics = self._get_physics(obj)
-
-        if physics == "flow":
-            return self.get_bc_val_flow()
-        elif physics == "transport":
-            return self.get_bc_val_transport()
-        elif physics == "mechanics":
-            return self.get_bc_val_mechanics()
-        else:
-            raise ValueError(
-                'Unknown physics "%s".\n Possible physics are: %s'
-                % (physics, self.known_physics)
-            )
-
-    def set_bc_val(self, obj, val):
-        """ Set physics-specific boundary condition
 
-        Parameters:
+def initialize_data(data, g, keyword, specified_parameters=None):
```

Comment:
I don't think I like this.I would rather remove this function and include this functionality to what is now called dicts.flow_dictinary, dicts.transport_dictionary, etc. 

The module should than probably be called default_parameters and have function called unit_flow_parameters. 

The syntax for the unit_flow will then be:
pp.unit_flow_parameters(g, keyword, Parameters=None, specified_parameters=None)
where if Parameters is None a  Parameter class is created, else, the unit flow parameters are added with the keyword "keyword". I'm not sure if we want to include specified_parameters or force the user to overwrite it himself.

Also we probably want a way to set parameters from a Rock class or a fluid class: